lightningd/options.c

```diff
@@ -449,6 +449,9 @@ static void setup_default_config(struct lightningd *ld)
 		ld->config = testnet_config;
 	else
 		ld->config = mainnet_config;
+
+        /* Set default PID file name to be per-network */
+        ld->pidfile = tal_fmt(NULL, "lightning-%s.pid", get_chainparams(ld)->network_name);
```

Comment:
Use `tal_fmt(ld, ...` so that owner of `ld->pidfile` string is freed when `ld` is freed. Otherwise CI will complain that this is a leak.

Reply:
Hi,

Yes, that line is causing a memory leak but using  `tal_fmt(ld, ...` instead of `tal_fmt(NULL` does not seem to fix it. Currently trying to find cause.

Reply:
Cause of memory leak found.

`setup_default_config(struct lightningd *ld)` is run twice in `lightningd/options.c` meaning `tal_fmt(ld, ...` is run a second time without a `tal_free` of first run.